overlord/fdestate/fdemgr.go

```diff
@@ -472,6 +472,53 @@ func GetKeyslots(st *state.State, keyslotRefs []KeyslotRef) (keyslots []Keyslot,
 	return mgr.GetKeyslots(keyslotRefs)
 }
 
+// NextUniqueKeyslot returns a unique keyslot reference given a
+// container role and a prefix which  can be used for conflict
+// free ephemeral key names. This can be used for temporary
+// key names during key replacement or re-provisioning.
+// The returned keyslot is only unique within the specified
+// container.
+//
+// The state needs to be locked by the caller.
+func (m *FDEManager) NextUniqueKeyslot(containerRole, prefix string) (ref KeyslotRef, err error) {
+	var reservedKeyslots map[string]bool
+
+	type fdeReservedKeyslots struct{}
+	cached := m.state.Cached(fdeReservedKeyslots{})
+	if cached == nil {
+		keyslots, _, err := m.GetKeyslots(nil)
+		if err != nil {
+			return KeyslotRef{}, fmt.Errorf("internal error: cannot find a unique keyslot for container role %q with prefix %q: %v", containerRole, prefix, err)
+		}
+		reservedKeyslots = make(map[string]bool, len(keyslots))
+		for _, keyslot := range keyslots {
+			reservedKeyslots[keyslot.Ref().String()] = true
+		}
+	} else {
+		reservedKeyslots = cached.(map[string]bool)
+	}
+
+	// We should run out of real LUKS2 keyslots ages before we hit this
+	// artificial limit. Realistically LUKS2 allows a maximum of 32 keyslots,
+	// and since we are aggressively reusing available postfix numbers, all
+	// should be good.
```

Comment:
I would use the fact differently and if we reach i == 32 or so, reread things from disk instead of continuing using improbable large numbers

Reply:
to be clear I expect we would read at most once per call to this function, but if we reach i == 32 or maybe 64/128? then we re-read the disk and start checking from 1 and then continue, we don't really need a limit on the number if we do it that way